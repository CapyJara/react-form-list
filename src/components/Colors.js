import React from 'react';
import PropTypes from 'prop-types';
import Color from './Color';

function Colors({ colors }) {
  return colors.map(({ name, hex, rgb }) => (
    <li key={`${name}-${hex}-${rgb}`}>
      <Color name={name} hex={hex} rgb={rgb}/>
    </li>
  ));
}

Colors.prototype = {
  colors: PropTypes.array.isRequired
};

export default Colors;
