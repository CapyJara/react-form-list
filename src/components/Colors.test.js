import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors';

describe('Colors component', () => {
  it('renders a Colors', () => {
    const wrapper = shallow(
      <Colors colors={[{ name: 'pink', hex: 'bl234' }, { name: 'pink', hex: 'bl234' }]}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
