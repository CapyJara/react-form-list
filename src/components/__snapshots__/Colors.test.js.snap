// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors component renders a Colors 1`] = `
<ul>
  <li
    key="pink"
  >
    <Color
      hex="bl234"
      name="pink"
    />
  </li>
  <li
    key="pink"
  >
    <Color
      hex="bl234"
      name="pink"
    />
  </li>
</ul>
`;
